agent/node.go

```diff
@@ -78,21 +78,23 @@ type NodeConfig struct {
 // cluster. Node handles workloads and may also run as a manager.
 type Node struct {
 	sync.RWMutex
-	config        *NodeConfig
-	remotes       *persistentRemotes
-	role          string
-	roleCond      *sync.Cond
-	conn          *grpc.ClientConn
-	connCond      *sync.Cond
-	nodeID        string
-	started       chan struct{}
-	stopped       chan struct{}
-	ready         chan struct{}
-	closed        chan struct{}
-	err           error
-	agent         *Agent
-	manager       *manager.Manager
-	roleChangeReq chan api.NodeRole
+	config               *NodeConfig
+	remotes              *persistentRemotes
+	role                 string
+	roleCond             *sync.Cond
+	conn                 *grpc.ClientConn
+	connCond             *sync.Cond
+	nodeID               string
+	nodeMembership       api.NodeSpec_Membership
+	started              chan struct{}
+	stopped              chan struct{}
+	ready                chan struct{}
+	certificateRequested chan struct{}
+	closed               chan struct{}
+	err                  error
+	agent                *Agent
+	manager              *manager.Manager
+	roleChangeReq        chan api.NodeRole
```

Comment:
No, these updates come from agent when the dispatcher api connection has been established. We want to get the membership status as soon as we have requested a certificate so we will know if we will block because of the pending state.